crates/host/src/wasmbus/providers/mod.rs

```diff
@@ -1,2 +1,359 @@
+                                            ?provider_id,
+                                            "failed to publish provider health check failed event",
+                                        );
+                                    }
+                                }
+                                // If the provider health status didn't change, we simply publish a health check status event
+                                (Ok(_), _) => {
+                                    if let Err(e) = event::publish(
+                                        &event_builder,
+                                        &ctl_nats,
+                                        &lattice,
+                                        "health_check_status",
+                                        event::provider_health_check(
+                                            &host_id,
+                                            &provider_id,
+                                        )
+                                    ).await {
+                                        warn!(
+                                            ?e,
+                                            ?provider_id,
+                                            "failed to publish provider health check status event",
+                                        );
+                                    }
+                                },
+                                _ => warn!(
+                                    ?provider_id,
+                                    "failed to deserialize provider health check response"
+                                ),
+                            }
+                        }
+                        else {
+                            warn!(?provider_id, "failed to request provider health, retrying in 30 seconds");
+                        }
+                }
+                exit = exit_health_rx.recv() => {
+                    if let Err(err) = exit {
+                        warn!(%err, ?provider_id, "failed to receive exit in health check task");
+                    }
+                    break;
+                }
+            }
+        }
+    }
+}
+
+/// Watch for config updates and send them to the provider
+///
+/// Returns a future that continually checks provider config changes
+/// until the config receiver gets a message
+fn watch_config(
+    rpc_nats: Arc<Client>,
+    config: Arc<RwLock<ConfigBundle>>,
+    lattice: Arc<str>,
+    provider_id: String,
+    mut exit_config_rx: broadcast::Receiver<()>,
+) -> impl Future<Output = ()> {
+    let subject = provider_config_update_subject(&lattice, &provider_id);
+    trace!(?provider_id, "starting config update listener");
+    async move {
+        loop {
+            let mut config = config.write().await;
+            select! {
+                maybe_update = config.changed() => {
+                    let Ok(update) = maybe_update else {
+                        // TODO: shouldn't this be continue?
```

Comment:
I think this should probably at least log something. This would mean there was an error when listening for the config changed event

Reply:
I'd like to address these config TODOs in a followup, but I definitely wanted to call this out because it felt weird when I was reading over it